src/components/views/settings/tabs/user/EncryptionUserSettingsTab.tsx

```diff
@@ -189,8 +191,14 @@ function useCheckEncryptionState(state: State, setState: (state: State) => void)
 
     useTypedEventEmitter(matrixClient, ClientEvent.AccountData, (event: MatrixEvent): void => {
         const type = event.getType();
-        // Recheck the status if this account data has been updated as this implies it has changed
-        if (type === "m.org.matrix.custom.backup_disabled") {
+        // Recheck the status if this account data has been updated as this implies the status
+        // of megolm key backup has changed on the user's account (there's no event emitted for megolm
+        // key backup enabled state changing, so we use this instead).
```

Comment:
> there's no event emitted for megolm key backup enabled state changing

I don't think that's actually true: isn't `CryptoEvent.KeyBackupStatus` what you need?

Reply:
Yes, not sure why I didn't spot that before. Changed it to use that and all seems to be working nicely.